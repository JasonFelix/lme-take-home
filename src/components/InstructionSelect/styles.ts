import styled from "styled-components";

const InstructionSelectContainer = styled.div`
  height: 300px;
  width: 100%;
  box-sizing: border-box;
  counter-reset: instruction-list;
  user-select: none;
`;

const Select = styled.ol`
  height: 284px;
  overflow-y: auto;
  padding: 0;
`;

const Option = styled.div`
  width: 100%;
  text-align: left;
  cursor: pointer;
  display: flex;
  flex-direction: row;

  &:before {
    counter-increment: instruction-list; /* Increment the value of section counter by 1 */
    content: counter(instruction-list) ". ";
    width: 32px;
    text-align: right;
    align-items: center;
    justify-content: center;
    display: flex;
  }

  &:hover {
    &:after {
      content: "remove";
      height: 100%;
      flex: 1;
      text-align: right;
      color: red;
    }
  }
`;

export { InstructionSelectContainer, Select, Option };
