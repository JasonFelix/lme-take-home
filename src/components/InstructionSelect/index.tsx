import { Empty, List } from 'antd';

import { Instruction } from "../../models/types";
import { InstructionSelectContainer, Option } from "./styles";

const INSTRUCTION_MAP = {
  'L': '\u2190 Turn Left',
  'R': '\u2192 Turn Right',
  'F': '\u2191 Walk Forward',
};

const InstructionSelect = ({ instructions, callback }: { instructions: Instruction[], callback: (index: number) => void }) => {
  const data = instructions.map((instruction, index) => 
    <List.Item key={index + instruction} onClick={() => callback(index)}>{INSTRUCTION_MAP[instruction]}</List.Item>)
  return <InstructionSelectContainer>
        {
          instructions.length
            ? <List 
                bordered
                dataSource={data} 
                size="small"
                style={{overflowY: 'scroll', height: '100%'}}
                renderItem={item => <Option>{item}</Option>}
              />
            : <Empty description={<span>No instructions</span>
            } />
        }

  </InstructionSelectContainer>
}

export default InstructionSelect;