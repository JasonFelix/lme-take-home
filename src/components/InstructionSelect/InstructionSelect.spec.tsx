import renderer from "react-test-renderer";
import InstructionSelect from ".";

it("InstructionSelect renders correctly", () => {
  const tree = renderer
    .create(<InstructionSelect instructions={["F"]} callback={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
