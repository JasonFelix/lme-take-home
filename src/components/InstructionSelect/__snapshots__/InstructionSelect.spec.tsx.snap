// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructionSelect renders correctly 1`] = `
<div
  className="sc-bdfBQB ktyAFe"
>
  <div
    className="ant-list ant-list-sm ant-list-split ant-list-bordered"
    style={
      Object {
        "height": "100%",
        "overflowY": "scroll",
      }
    }
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div
        className="ant-spin-container"
      >
        <ul
          className="ant-list-items"
        >
          <div
            className="sc-dlfnuX kGVqiQ"
          >
            <li
              className="ant-list-item"
              onClick={[Function]}
            >
              â†‘ Walk Forward
            </li>
          </div>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
